File: src/currency.py
"""This module contain Currency class."""

from .db import get_attribute_by_id, get_id_by_attribute


class Currency:

    def __init__(self):
        self.id = 0
        self.name = ''
        self.symbol = ''

    def get_currency_by_id(self):
        self.name = get_attribute_by_id(self, 'name')
        self.symbol = get_attribute_by_id(self, 'symbol')
        return self

    def get_currency_by_name(self):
        self.id = get_id_by_attribute(self, 'name')
        self.get_currency_by_id()
        return self

    def __str__(self):
        return '\'' + self.name + '\', \'' + self.symbol + '\''

    def __dir__(self):
        return ['name', 'symbol']

File: Sources/np/Models/Currency.swift
//
//  Currency.swift
//  
//
//  Created by Boris Ovodov on 21.05.2024.
//

import Fluent
import Vapor

final class Currency: Model, @unchecked Sendable, Content {
    static let schema = "currencies"
    
    @ID(key: .id)
    var id: UUID?

    @Field(key: "name")
    var name: String
    
    @Field(key: "symbol")
    var symbol: String
    
    @Field(key: "code")
    var code: String
    
    @Boolean(key: "isSymbolLeftFromValue")
    var isSymbolLeftFromValue: Bool
    
    init() { }
}

extension Currency: CustomStringConvertible {
    var description: String {
        self.name
    }
}
